### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by mikesch803**
1. add guest login
2. refactor the code
3.  fix logout handler

### CODE REVIEW

File: src/context/playlist-context.jsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ const PlaylistProvider = ({ children }) => {
           }
         );
         if (response.status === 201) {
-          dispatch({ type: "PLAYLISTS", payload: response.data.playlists });
+          playlistDispatch({ type: "PLAYLISTS", payload: response.data.playlists });
```
Comment (MayankKumar10) [inline src/context/playlist-context.jsx:39]: you can use here destructure for playlist

File: src/pages/liked-videos-page/LikedVideos.jsx [hunk 2]
```diff
@@ -19,11 +19,13 @@ export function LikedVideos() {
         });
         if (response.status === 200) {
           setLikedVideos(response.data.likes);
+          console.log(response)
         }
       } catch (err) {
         console.error(err);
```
Comment (MayankKumar10) [inline src/pages/liked-videos-page/LikedVideos.jsx:25]: remove console log

**Review by MayankKumar10 — APPROVED** [review]
Look into these Suggestions Everything else looks good to me you can merge it.

File: src/context/auth-context.jsx [hunk 3]
```diff
@@ -99,9 +106,37 @@ const AuthProvider = ({ children }) => {
...
+
   const logoutHandler = () => {
-    localStorage.clear();
+    setHistoryVideos([]);
+    setLikedVideos([]);
+    setWatchLaterVideos([]);
+    playlistDispatch({ type: "RESET_ALL_PLAYLIST" });
+    playlistDispatch({ type: "DELETE_PLAYLIST" });
     setUserState(false);
+    localStorage.clear();
```
Comment (anik31) [inline src/context/auth-context.jsx:139]: appreciation - you have implemented login & logout perfectly

File: src/context/like-video-context.jsx [hunk 4]
```diff
@@ -27,7 +28,7 @@ const LikedVideoProvider = ({ children }) => {
         if (response.status === 201) {
           setLikedVideos(response.data.likes);
           setToastStyles("alert alert-info")
-          setToastMsg("video is added to liked videos")
+          setToastMsg("Added to liked videos")
           setToastState(true);
```
Comment (anik31) [inline src/context/like-video-context.jsx:32]: appreciation - good that you're showing toast

**Review by anik31 — APPROVED** [review]
Everything looks good @mikesch803

**Pull request merged**